.github/workflows/test-all-packages.yml

```diff
@@ -80,6 +80,51 @@ jobs:
     - name: lint check
       run: yarn lint-check
 
+  dev-canary:
+    # FIXME: if: ${{github.event_name == 'push'}}
+    needs: build
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        # note: only use one node-version
+        node-version: ['14.x']
+    steps:
+    - uses: actions/setup-node@v1
+      with:
+        node-version: ${{ matrix.node-version }}
+    # BEGIN-RESTORE-BOILERPLATE
+    - name: restore built files
+      id: built
+      uses: actions/cache@v1
+      with:
+        path: .
+        key: ${{ runner.os }}-${{ matrix.node-version }}-built-${{ github.sha }}
+    - uses: actions/checkout@v2
+      with:
+        submodules: 'true'
+      if: steps.built.outputs.cache-hit != 'true'
+    - name: yarn install
+      run: yarn install
+      if: steps.built.outputs.cache-hit != 'true'
+    - name: yarn build
+      run: yarn build
+      if: steps.built.outputs.cache-hit != 'true'
+    # END-RESTORE-BOILERPLATE
+
+    # Adapted from https://johnny.sh/notes/publish-canary-lerna-cicd/
+    - name: configure NPM token
+      run: |
+        echo "//registry.npmjs.org/:_authToken=${NPM_TOKEN}" > ~/.npmrc
```

Comment:
`NPM_TOKEN` looks like a _very_ pointy instrument: it seems to represent authority to update all our packages in npm.

I'd like to get @warner 's take on this.

Reply:
Please move this discussion to #4120.